front/src/modules/simulationResult/components/SpaceTimeChart/useLazyProjectTrains.ts

```
@@ -0,0 +1,145 @@
+  const projectionSeqNum = useRef(0);
+
+  const [postV2TrainScheduleProjectPath] =
+    osrdEditoastApi.endpoints.postV2TrainScheduleProjectPath.useLazyQuery();
+
+  const trainSchedulesById = useMemo(() => mapBy(trainSchedules, 'id'), [trainSchedules]);
+
+  // gradually project the trains on the selected path
+  useEffect(() => {
+    const projectNextPackage = async (
+      _path: PathfindingResultSuccess,
+      packageToProject: number[]
+    ) => {
+      packageToProject.forEach((trainId) => requestedProjectedTrainIds.current.add(trainId));
+
+      const { blocks, routes, track_section_ranges } = _path;
+      const rawProjectedTrains = await postV2TrainScheduleProjectPath({
+        projectPathForm: {
+          infra_id: infraId!,
+          ids: packageToProject,
+          path: { blocks, routes, track_section_ranges },
+          electrical_profile_set_id: electricalProfileSetId,
+        },
+      }).unwrap();
+
+      setProjectedTrainsById((prevTrains) => {
+        const newProjectedTrains = upsertNewProjectedTrains(
+          prevTrains,
+          rawProjectedTrains,
+          trainSchedulesById
+        );
+        return newProjectedTrains;
+      });
+      // console.log('2. package', packageToProject);
+    };
+
+    const projectTrains = async (
+      seqNum: number,
+      _path: PathfindingResultSuccess,
+      _trainToProjectIds: number[]
+    ) => {
+      const shouldProjectIds = _trainToProjectIds.filter(
+        (trainId) => !requestedProjectedTrainIds.current.has(trainId)
+      );
+
+      for (let i = 0; i < shouldProjectIds.length; i += BATCH_SIZE) {
+        // If projection parameters have changed, bail out
+        if (projectionSeqNum.current !== seqNum) break;
+
+        const packageToProject = getBatchPackage(i, shouldProjectIds, BATCH_SIZE);
+        try {
+          await projectNextPackage(_path, packageToProject);
+        } catch (e) {
+          console.error('error', e);
```

Can we bubble up this error to the user, instead of keeping it in the console where only developers know where to look? Can be confusing to look at a blank page which never gets populated without any indication that something failed.

---

Yes, I added a notification :+1: